tests/test_pds4_create_xml_index_blackbox.py

```diff
@@ -15,21 +15,67 @@
 
 
 @pytest.mark.parametrize(
-        'golden_file,new_file,cmd_line',
+        'golden_file,new_file_index,new_file_headers,cmd_line',
         [
+            # Executable command: pds4_create_xml_index ../test_files/labels "tester_label_1.xml"
+            (
+                str(expected_dir / 'index_file_success.csv'),
+                None, None,
+                []
+            ),
+
+            # Executable command: pds4_create_xml_index ../test_files/labels "tester_label_1.xml" --generate-label ancillary
+            (
+                str(expected_dir / 'index_file_success.csv'),
+                None, None,
+                [
+                    '--generate-label',
+                    'ancillary'
+                 ]
+            ),
+
+            # Testing --limit-xpaths-file with two outputs
+            # Executable command: pds4_create_xml_index ../test_files/labels "tester_label_1.xml" --limit-xpaths-file ../test_files/samples/element_1.txt --output-headers-file limit_xpaths_file.txt --output-index-file limit_xpaths_file.csv
+            # Compare result to golden copy:
+            # test_files/expected/limit_xpaths_file_success_1.txt
+            (
+                str(expected_dir / 'limit_xpaths_file_success_1.csv'),
+                'limit_xpaths_file.csv', 'limit_xpaths_file.txt',
+                [
+                    str(test_files_dir),
+                    str(labels_dir.name / Path('tester_label_1.xml')),
```

Comment:
Because the second part is the pattern, which needs to be relative to the provided directory. Absolute paths given as patterns cause the program to crash.

Reply:
Ah, right, of course. That said, it would better to have a global variable called `LABELS_DIR_NAME` and you can still eliminate the extra calls to `Path()`.